app/src/main/res/layout/fragment_search_result.xml

```diff
@@ -4,37 +4,75 @@
     android:layout_height="match_parent">
 
-    <TextView
-        android:id="@+id/tvResults"
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:id="@+id/resultFoundLayout"
         android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:layout_marginBottom="@dimen/margin_standard_half"
-        android:background="@color/colorAccent"
-        android:padding="@dimen/margin_standard_half"
-        android:text="Results for "
-        android:textColor="@android:color/white"
-        android:textSize="@dimen/text_size"
-        app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+        android:layout_height="match_parent"
+        android:visibility="gone">
 
-    <View
-        android:id="@+id/lineView"
-        android:layout_width="wrap_content"
-        android:layout_height="1dp"
-        android:background="@android:color/white"
-        app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@id/tvResults" />
+        <TextView
+            android:id="@+id/tvResults"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginBottom="@dimen/margin_standard_half"
+            android:background="@color/colorAccent"
+            android:padding="@dimen/margin_standard_half"
+            android:text="Results for "
+            android:textColor="@android:color/white"
+            android:textSize="@dimen/text_size"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
 
-    <androidx.recyclerview.widget.RecyclerView
-        android:id="@+id/resultsList"
+        <View
+            android:id="@+id/lineView"
+            android:layout_width="wrap_content"
+            android:layout_height="1dp"
+            android:background="@android:color/white"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/tvResults" />
+
+        <androidx.recyclerview.widget.RecyclerView
+            android:id="@+id/resultsList"
+            android:layout_width="@dimen/match_constraint"
+            android:layout_height="@dimen/match_constraint"
+            app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@id/lineView" />
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+    <androidx.constraintlayout.widget.ConstraintLayout
```

Comment:
It is quite inefficient to use nested `ConstraintLayout`'s. One of the main purposes of `ConstraintLayout` was to flatten the layout structure, so that we don't have those nested structures. This `ConstraintLayout` can be wasily replaced with just one TextView with drawableTop parameter. Then you don't need any nested `ConstraintLayout`'s any more.